app/components/BuynBTC.tsx

```diff
@@ -0,0 +1,126 @@
+import { Card, CardContent } from "./ui/card";
+import { Button } from "./ui/button";
+import { FormInput } from "./form/FormInput";
+import { FormProvider, useForm } from "react-hook-form";
+import { useContext } from "react";
+import { WalletContext } from "~/providers/ByieldWalletProvider";
+import { ByieldWallet } from "~/types";
+import { SuiModal } from "./Wallet/SuiWallet/SuiModal";
+import { useSignAndExecuteTransaction, useSuiClient } from "@mysten/dapp-kit";
+import { Transaction } from "@mysten/sui/transactions";
+import { useNetworkVariables } from "~/networkConfig";
+import { suiToMist } from "~/util/util";
+import { useToast } from "~/hooks/use-toast";
+
+interface ExchangeRateProps {
+	youReceive: number;
+}
+
+function Fee({ youReceive }: ExchangeRateProps) {
+	return (
+		<Card className="p-4 bg-azure-10 rounded-2xl h-14">
+			<CardContent className="flex flex-col justify-between p-0">
+				<div className="flex justify-between gap-2">
+					<p className="text-gray-400">You Receive</p>
+					<p>{youReceive} nBTC</p>
+				</div>
+			</CardContent>
+		</Card>
+	);
+}
+
+interface OtcBuyForm {
+	numberOfSuiCoins: number;
+}
+
+export function BuynBTC() {
+	const { toast } = useToast();
+	const { connectedWallet } = useContext(WalletContext);
+	const client = useSuiClient();
+	const { packageId, module, swapFunction, vaultId, pricePerNBTCInSUI } = useNetworkVariables();
+	const isSuiWalletConnected = connectedWallet === ByieldWallet.SuiWallet;
+	const { mutate: signAndExecuteTransaction } = useSignAndExecuteTransaction({
+		execute: async ({ bytes, signature }) =>
+			await client.executeTransactionBlock({
+				transactionBlock: bytes,
+				signature,
+				options: {
+					// Raw effects are required so the effects can be reported back to the wallet
+					showRawEffects: true,
+					// Select additional data to return
+					showObjectChanges: true,
+				},
+			}),
+	});
+
+	const otcBuyForm = useForm<OtcBuyForm>();
+	const { watch } = otcBuyForm;
+	const numberOfSuiCoins = watch("numberOfSuiCoins");
```

Comment:
```suggestion
	const AmountOfSUI = watch("numberOfSuiCoins");
```

Reply:
btw. `SUI` is the coin, `Sui` is the blockchain

Reply:
maybe `suiAmount` is the best here

Reply:
I replaced it with `suiAmount`